### ISSUE TITLE
Still unable to use stackoverflow.com_en_all_2019-02.zim

### ISSUE THREAD

**Issue by CrushedPixl** [openzim/zim-tools#210]
Yesterday's release included [this](https://github.com/openzim/libzim/pull/460) pull request, which I assumed would fix [this](https://github.com/kiwix/kiwix-tools/issues/413) issue as was suggested [here](https://github.com/openzim/zim-tools/issues/126), but the issue remains for me.

With Parabola's (x86) packages I get
```
$ zimcheck -v
2.1.0

$ kiwix-serve -V
3.1.2

$ zimcheck stackoverflow.com_en_all_2019-02.zim
[INFO] Checking zim file stackoverflow.com_en_all_2019-02.zim
[INFO] Verifying ZIM-file structure integrity...
Error getting mimelists.
  [ERROR] ZIM file's low level structure is invalid
Error getting mimelists.

$ kiwix-serve -vp 8080 stackoverflow.com_en_all_2019-02.zim
Unable to add the ZIM file 'stackoverflow.com_en_all_2019-02.zim' to the internal library.

$ sha256sum stackoverflow.com_en_all_2019-02.zim # Matches
224b8a368d381b8f71c04c0413e2e5bde586369fc86a57b9622ba7a070b8ee47  stackoverflow.com_en_all_2019-02.zim

# Testing that a non-problematic file works
$ kiwix-serve -vp 8080 wikipedia_en_all_nopic_2020-10.zim # Website works
```

And with my (most likely flawed) self-built (x86) binaries (through kiwix-build, built today)
```
$ /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/zimcheck -V
2.1.0

$ /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/kiwix-serve -V
3.1.2

# This fails with an identical error using a non-problematic file
$ /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/zimcheck stackoverflow.com_en_all_2019-02.zim 
[INFO] Checking zim file stackoverflow.com_en_all_2019-02.zim
[INFO] Verifying ZIM-archive structure integrity...
/tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/zimcheck: symbol lookup error: /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/zimcheck: undefined symbol: _ZN3zim8validateERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt6bitsetILm5EE

$ /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/kiwix-serve -vp 8080 stackoverflow.com_en_all_2019-02.zim
Unable to add the ZIM file 'stackoverflow.com_en_all_2019-02.zim' to the internal library.

# Testing that a non-problematic file works
$  /tmp/a/kiwix-build/BUILD_native_dyn/INSTALL/bin/kiwix-serve -vp 8080 wikipedia_en_all_nopic_2020-10.zim # Website works
```

**Comment by kelson42** [comment]
@CrushedPixl You have been fooled by a "fake" Github release. This released has already been done months ago, therefore the PR you talk about is not in it. We plan to make a new libzim release next week.

**Comment by EnKrypt** [comment]
Hi, is there any update on this? There seems to be no open issue being tracked at the moment, but this problem is still present. I don't see a newer release of `kiwix-serve`.

At least one issue should remain open until a release specifically closes it.
Please tell us what we, the community can do to help speed this up, because I'm pretty sure folks will just keep coming along and opening a new ticket every month if they find tickets closed but the problem still persisting.

We are here to help, many of us are devs and are happy to contribute. 

Gotta feel sorry for folks who download 135GiB and then find this thread after `kiwix-serve` wouldn't run.

**Comment by EnKrypt** [comment]
@kelson42 Can this issue be re-opened? A humble request so that I can track the progress of this until a release is actually out that actually closes this issue.

I can create a new ticket if you'd prefer.

**Comment by mgautierfr** [comment]
> Please tell us what we, the community can do to help speed this up, because I'm pretty sure folks will just keep coming along and opening a new ticket every month if they find tickets closed but the problem still persisting.
> We are here to help, many of us are devs and are happy to contribute

We are stuck we a exceptionally big release of libzim which need a bit more work to be finalized/stabilized.
There is nothing you can do to speed up that.
However, if you want to help, you are more than welcome. There are a lot of issues (easy or not) in all our repositories (https://github.com/kiwix and https://github.com/openzim organizations)

> At least one issue should remain open until a release specifically closes it.

We close issues when there is nothing more to do on them.
It would be indeed nice to have a status telling if the issue is fixed on master only or if it is released.
It is a change in our process and we have to discuss with @kelson42 about this.

**Comment by jmontleon** [comment]
If anyone is trying to overcome this, [this](https://github.com/jmontleon/fedora-kiwix-server) is an example of how you can build your own kiwix-serve that is capable of serving the stackoverflow zim.

**Comment by mgautierfr** [comment]
For your information https://github.com/kiwix/kiwix-build works pretty well on fedora.

**Comment by jmontleon** [comment]
This assert is failing trying to build a new stackoverflow:
https://github.com/openzim/libzim/blob/master/src/writer/creator.cpp#L334

I think this means the mimelist is still coming up too big?

I see on https://github.com/openzim/zim-tools/issues/126 that the list of mimetypes garbage like this:

```
  "-%3C%3E%5BPromotion%5D,%20%5BItem%5D-%3C%3E%5BPromotedPurchaseOption%5D",
  "-%3E%5Bnull%5D,%5Bmain%5Dcircle_1-%3E%5BaCircle%2810%29%5D,%5BaCircle%2810%29%5Dcenter-%3E%5BaPoint%287%208%29%5D,%5BaPoint%285%205%29%5D",
```

These are coming from file extensions on links I believe:
```
output/question/3344350:<p><strong>Breaking the <code>PurchaseOption</code>&lt;-&gt;<code>Promotion</code> HABTM association and creating a new one for <code>Item</code>&lt;-&gt;<code>PromotedPurchaseOption</code></strong></p>

output/question/3344350:<p><img src="https://yuml.me/diagram/scruffy/class/%5BItem%5D-%3C%3E%5BPurchaseOption%5D,%20%5BPurchaseOption%5D%3C%3E-%5BPromotedPurchaseOption%5D,%20%5BPromotedPurchaseOption%5D-%3C%3E%5BPromotion%5D,%20%5BItem%5D-.-%3C%3E%5BPromotion%5D,%20%5BItem%5D-%3C%3E%5BPromotedPurchaseOption%5D" alt=""></p>

output/question/3714146:<p><img src="https://i.stack.imgur.com/ZGlcx.jpgBlockquote" alt="alt text"></p>
```

If the logic is to call everything after a dot the mimetype I think this will remain a problem.

**Comment by jmontleon** [comment]
![image](https://user-images.githubusercontent.com/1850500/122149878-c4c31680-ce2a-11eb-9bbf-ce8b67b6e00a.png)

`-rw-r--r--. 1 root  root  146G Jun 15 22:35 stackoverflow.com_en_all.zim`

I was able to complete a working build off of the 2021-03 dump. I'm probably going to start working on one of hte 2021-06 dump. I ended up using sotoki, some scripts, and zimwriterfs. sotoki 1.3.2-dev was hopelessly slow writing. My file failed to build because the mimetype list was far far too big (even bigger than the 2019 one).

I had to workaround that as described here: https://github.com/openzim/zim-tools/issues/187

Only thing I noticed is I failed to update the tag links on index page (although the all tags link and links from there work). I'll fix that on the next go around.
